git/README.md

```diff
@@ -84,44 +92,46 @@ Used for asking questions or delivering new information.
 
 ## Commits
 
-Indiviual commits should be [atomic changes](https://en.wikipedia.org/wiki/Atomic_commit) to the codebase.
+Individual commits should be [atomic changes](https://en.wikipedia.org/wiki/Atomic_commit) to the codebase.
 
-- A commit should not leave the code in a broken state waiting on a later commit to fix it. These commits are fine in development stages but *have* to be squashed before pulling into the master branch.
-- Changes that are not related should be split into separate commits. This must be taken with the previous rule in mind to help determine what's related or not. Bugfixes that are prerequisites to your change but not the main point of the change should be made in separate commits that come first.
-- Make sure changes are rebased on latest master. Ensure your code is up-to-date to prevent merge conflicts.
+- A commit should not leave the code in a broken state waiting on a later commit to fix it. These commits are fine in development stages but *have* to be squashed before pulling into `master`.
```

Comment:
I feel like: `A commit should not leave the code in a broken state waiting on a later commit to fix it`. Is contradicted by the next line: `these commits are fine in development stages...`. If you can commit broken code then it should start with that and then explain when you can't like when you request your PR to be reviewed.

Reply:
It's a good point. You're right, *most* commits leave the code in a broken state, hence why we squash them before pulling into master (why have 10 semi-implemented broken functionality commits in `master`?). The point I am trying to make is about commits in `master`, each should be atomic in the sense that each commit is a valid state that can be rolled to. While developing, you'll get a lot of "Minor fix" "Fix linting" etc, commits. We don't care about the fact that the code was "broken" previous to these commits, because it'd be counterproductive to squash all of these (causing more headache and time than otherwise).

How about 
"- While developing commits can contain broken code (failing tests, unlinted, broken functionality), as it would be too onerous and counterproductive to expect every commit to be fully atomic. Each commit in `master`, however, fully implements a functionality (barring unknown bugs), has a fully passing test suite, and can be built and deployed with every expectation that the code will function accordingly. All commits that do not meet these requirements are squashed before being pulled into `master`"

Reply:
Yeah, I like that.